### ISSUE TITLE
Runtime errors when running test suite

### ISSUE THREAD

**Issue by Tenshi65535** [fourmond/QSoas#3]
Greetings to all, this is the AUR package maintainer of QSoas.

QSoas with Mruby 3 can be build successfully and here are the patches applied.
```bash
sed -r -i "s:MRUBY_RELEASE_MAJOR == 2:MRUBY_RELEASE_MAJOR >= 2:g" ./src/mruby.cc
#Enable Mruby3
sed -r -i "s:struct mrb_irep \*:const &:g" ./src/mruby.cc
#mrb_irep is constified
sed -r -i "s:lv\[i\]\.name:lv\[i\]:g" ./src/mruby.cc
#lv is no more a class
sed -r -i "433,440s:.:\/\/&:" ./src/mruby.cc
#Remove deprecated OP code
sed -r -i "500,501s:.:\/\/&:" ./src/mruby.cc
#Remove deprecated OP code
sed -r -i "505,506s:.:\/\/&:" ./src/mruby.cc
#Remove deprecated OP code
sed -r -i "s:OP_RAISE:OP_RAISEIF:g" ./src/mruby.cc
#Replace renamed OP code
```

I've tried the test suite but some runtime error occurred. Here is the output from the terminal:
```bash
This is QSoas version 3.1 running with mruby 3.1.0 and Qt 5.15.6
Built 2022-11-10 12:34:07 +0900 on 'cp2kenv' with Qt 5.15.6 and GSL version 2.7.1
 * application dir: /usr/bin
 * application path: /usr/bin/qsoas
 * library paths: 
    - /usr/lib/qt/plugins
    - /usr/bin
 * documentation file: 
Not using libzip
Starting timer

catalytic-waves: ...........................
columns: .....................F: $stats["x_max"] == 130
F: $stats["x_max"] == 140
F: $stats["x_max"] == 150
....F: $stats["y_average"] - 20 (should be below: 1e-09 but is: -20)
F: $stats["y2_average"] - 19 (should be below: 1e-09 but is: -19)
F: $stats["y8_average"] - 13 (should be below: 1e-09 but is: -13)
.F: $stats["y_average"] - 10 (should be below: 1e-09 but is: -10)
.....................................................
fits: ..............................................F: $residuals - 0.02 (should be below: 1e-08 but is: 7.53716)
F: $residuals - 0.1 (should be below: 1e-08 but is: 7.45716)
F: $residuals**2 - 0.5*(0.02**2+0.1**2) (should be below: 1e-08 but is: 57.1055)
F: $residuals**2 - (100*0.02**2+0.1**2)/101 (should be below: 1e-08 but is: 57.1102)
F: $residuals**2 - (0.02**2+100*0.1**2)/101 (should be below: 1e-08 but is: 57.1008)
.F: $stats["y2_max"]-4 (should be below: 0 but is: -4)
F: $stats["y3_max"]-3 (should be below: 0 but is: 0.33333)
F: $stats["y4_max"]-2 (should be below: 0 but is: -2)
....F: $stats.y12_max (should be below: 1e-10 but is: nan)
Warning: QXcbConnection: XCB error: 3 (BadWindow), sequence: 3053, resource id: 8564473, major code: 40 (TranslateCoords), minor code: 0 (:0, )
....F: $stats.y12_max (should be below: 1e-10 but is: nan)
Warning: QXcbConnection: XCB error: 3 (BadWindow), sequence: 3698, resource id: 8565912, major code: 40 (TranslateCoords), minor code: 0 (:0, )
....F: $stats.y12_max (should be below: 1e-10 but is: nan)
....
formulas: ........F: $stats.y_max+2 (should be below: 0 but is: 2)
......................F: $stats['rows']==901
F: $stats['x_max']==9
F: $stats['rows']==2000
.............F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
.F: $stats["rows"] == 500
....F: $stats.x_max-4 (should be below: 0 but is: 6)
F: $stats.x_max-4 (should be below: 0 but is: 6)
F: $stats.y2_max-16 (should be below: 0 but is: -16)

functions: ......
globs: F: $stats.y_max-7 (should be below: 0 but is: -7)
F: $stats.y_max-6 (should be below: 0 but is: -6)
F: $stats.y_max-6 (should be below: 0 but is: -6)
F: $stats.y_max+4 (should be below: 0 but is: 4)
F: $stats.y_max+5 (should be below: 0 but is: 5)
F: $stats.y_max+6 (should be below: 0 but is: 6)
E: $stats.y_norm -- A ruby exception occurred: undefined method 'y_norm' (NoMethodError)
E: $meta.argument-1 -- A ruby exception occurred: undefined method 'argument' (NoMethodError)
E: $stats.y_norm -- A ruby exception occurred: undefined method 'y_norm' (NoMethodError)
E: $meta.argument-3 -- A ruby exception occurred: undefined method 'argument' (NoMethodError)
E: $stats.y_norm -- A ruby exception occurred: undefined method 'y_norm' (NoMethodError)
E: $meta.argument-2 -- A ruby exception occurred: undefined method 'argument' (NoMethodError)

inspect: .F: $stats["x_max"] - 2*2**3/(3.0) (should be below: 2e-05 but is: -5.33333)
.F: $stats["x_max"] - 2*2**3/(3.0) (should be below: 2e-05 but is: -5.33333)
....F: $stats["y_a"] - 2 (should be below: 1e-07 but is: -2)
F: $stats["y_b"] - 3 (should be below: 1e-07 but is: -3)
.F: $stats["rows"] == 1
F: $stats["y_max"] == 10
F: $stats["x_max"] == 0

integrators: ....terminate called after throwing an instance of 'InternalError'
  what():  Fatal error: ASSERT failure in QList<T>::operator[]: "index out of range", file /usr/include/qt/QtCore/qlist.h, line 579 (/usr/include/qt/QtCore/qlist.h:579, )
0x562d86c5f20a: qsoas(+0x22420a) [0x562d86c5f20a]
0x562d86c01bce: qsoas(+0x1c6bce) [0x562d86c01bce]
0x562d86c01c49: qsoas(+0x1c6c49) [0x562d86c01c49]
0x562d86ac9e9b: qsoas(+0x8ee9b) [0x562d86ac9e9b]
0x7fe3d2ce0a88: /usr/lib/libQt5Core.so.5(+0xe0a88) [0x7fe3d2ce0a88]
0x7fe3d2cd9b99: /usr/lib/libQt5Core.so.5(+0xd9b99) [0x7fe3d2cd9b99]
0x7fe3d2c9fec4: /usr/lib/libQt5Core.so.5(_ZNK14QMessageLogger5fatalEPKcz+0xb2) [0x7fe3d2c9fec4]
0x7fe3d2c9ff9d: /usr/lib/libQt5Core.so.5(+0x9ff9d) [0x7fe3d2c9ff9d]
0x562d86adf84e: qsoas(+0xa484e) [0x562d86adf84e]
0x562d86cf27ac: qsoas(+0x2b77ac) [0x562d86cf27ac]
0x562d86bc1ba5: qsoas(+0x186ba5) [0x562d86bc1ba5]
0x562d86bc7263: qsoas(+0x18c263) [0x562d86bc7263]
0x562d86b316c9: qsoas(+0xf66c9) [0x562d86b316c9]
0x562d86b32cfe: qsoas(+0xf7cfe) [0x562d86b32cfe]
0x562d86b3428b: qsoas(+0xf928b) [0x562d86b3428b]
0x562d86b35137: qsoas(+0xfa137) [0x562d86b35137]
0x562d86b274a6: qsoas(+0xec4a6) [0x562d86b274a6]
0x562d86b2d197: qsoas(+0xf2197) [0x562d86b2d197]
0x562d86b316c9: qsoas(+0xf66c9) [0x562d86b316c9]
0x562d86b32cfe: qsoas(+0xf7cfe) [0x562d86b32cfe]
Aborted (core dumped)
```
It seems there is something wrong with QT?

**Comment by fourmond** [comment]
Wow, that doesn't look good at all -- this looks like QSoas is severly crippled. My guess is that it's related to using mruby 3.1...

I transition very slowly to the newer versions since in general in involves many hours of tweaking and fixing.

I'll give it a try myself over the week, please ping me if you don't hear back.

Cheers,

**Comment by fourmond** [comment]
Release 3.2 unfortunately does not support yet building with mruby 3.1 and 3.2, but eec9d8c does, with errors in the test suite that reflect mostly inaccuracies rather than real issues.

**Comment by fourmond** [comment]
This is fixed in release 3.3. The accuracy with mruby 3.1 and 3.2 is not as good as with mruby 2.1 (linked to the boxing of floats), but it is nevertheless working fine.

**Comment by fourmond** [comment]
(closing with release 3.3)
